src/views/Admin/index.js

```diff
@@ -11,7 +12,11 @@ const Admin = ({ authentication }) => {
   const isOnline = useNetworkStatus();
 
   useEffect(() => {
-    setIsAdmin(user.getLocalInfo().college_role === 'god');
+    try {
+      setIsAdmin(user.getLocalInfo().college_role === 'god');
+    } catch (error) {
+      // Unauthorized exception
+    }
```

Comment:
You should specifically catch an unauthorized error here, and throw any other errors.
```suggestion
    } catch (error) {
      if (error instanceof AuthError) {
        // Unauthorized exception expected when user not authenticated
      } else {
        throw error;
      }
    }
```